Error: Illegal character in opaque part at index 47

```
java -jar jardiff.jar file1.jar file2.jar
java.lang.IllegalArgumentException: Illegal character in opaque part at index 47: jar:file:file1.jar
        at java.net.URI.create(Unknown Source)
        at scala.tools.jardiff.IOUtil$.rootPath(IOUtil.scala:16)
        at scala.tools.jardiff.JarDiff.$anonfun$diff$1(JarDiff.scala:35)
        at scala.collection.immutable.List.foreach(List.scala:312)
        at scala.tools.jardiff.JarDiff.renderAndCommit$1(JarDiff.scala:34)
        at scala.tools.jardiff.JarDiff.$anonfun$diff$3(JarDiff.scala:54)
        at scala.collection.Iterator$$anon$9.next(Iterator.scala:554)
        at scala.collection.Iterator$GroupedIterator.takeDestructively(Iterator.scala:205)
        at scala.collection.Iterator$GroupedIterator.go(Iterator.scala:220)
        at scala.collection.Iterator$GroupedIterator.fill(Iterator.scala:257)
        at scala.collection.Iterator$GroupedIterator.hasNext(Iterator.scala:261)
        at scala.collection.IterableOnceOps.foreach(IterableOnce.scala:576)
        at scala.collection.IterableOnceOps.foreach$(IterableOnce.scala:574)
        at scala.collection.AbstractIterator.foreach(Iterator.scala:1196)
        at scala.tools.jardiff.JarDiff.diff(JarDiff.scala:55)
        at scala.tools.jardiff.Main$.run(Main.scala:84)
        at scala.tools.jardiff.Main$.main(Main.scala:20)
        at scala.tools.jardiff.Main.main(Main.scala)
Caused by: java.net.URISyntaxException: Illegal character in opaque part at index 47: jar:file:file1.jar
        at java.net.URI$Parser.fail(Unknown Source)
        at java.net.URI$Parser.checkChars(Unknown Source)
        at java.net.URI$Parser.parse(Unknown Source)
        at java.net.URI.<init>(Unknown Source)
        ... 18 more
```

--- a/build.sbt
+++ b/build.sbt
@@ -35,7 +35,8 @@ lazy val core = (
       "org.eclipse.jgit" % "org.eclipse.jgit" % "5.3.1.201904271842-r",
       "org.slf4j" % "slf4j-api" % "1.7.26",
       "org.slf4j" % "log4j-over-slf4j" % "1.7.26", // for any java classes looking for this
-      "ch.qos.logback" % "logback-classic" % "1.2.3"
+      "ch.qos.logback" % "logback-classic" % "1.2.3",
+      "org.scalatest" %% "scalatest" % "3.1.1" % Test,
     ),
     name := buildName + "-core",
     headerLicense := Some(HeaderLicense.Custom("Copyright (C) Lightbend Inc. <https://www.lightbend.com>")),

--- a/core/src/main/scala/scala/tools/jardiff/IOUtil.scala
+++ b/core/src/main/scala/scala/tools/jardiff/IOUtil.scala
@@ -13,7 +13,7 @@ import java.util
 object IOUtil {
   def rootPath(fileOrZip: Path): Path = {
     if (fileOrZip.getFileName.toString.endsWith(".jar")) {
-      val uri = URI.create("jar:file:" + fileOrZip.toUri.getPath)
+      val uri = URI.create(s"jar:${fileOrZip.toUri}")
       newFileSystem(uri, new util.HashMap[String, Any]()).getPath("/")
     } else {
       val extSlash = ".jar/"

--- a/core/src/test/scala/scala/tools/jardiff/IOUtilSpec.scala
+++ b/core/src/test/scala/scala/tools/jardiff/IOUtilSpec.scala
@@ -1,0 +1,20 @@
+package scala.tools.jardiff
+
+import java.nio.file._
+import java.util.zip.ZipOutputStream
+
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+
+import scala.util.Using
+
+final class IOUtilSpec extends AnyFlatSpec with Matchers {
+  behavior of "IOUtil.rootPath"
+
+  it should "handle jar path with spaces" in {
+    val jar = Files.createTempDirectory("app support").resolve("best project.jar")
+    Using.resource(new ZipOutputStream(Files.newOutputStream(jar)))(_.closeEntry()) // create jar
+    IOUtil.rootPath(jar).toString shouldBe "/"
+    IOUtil.rootPath(jar.resolve("foo/Bar.class")).toString shouldBe "/foo/Bar.class"
+  }
+}

